### PR TITLE
Improved vpn error handling

### LINKED ISSUE #278: [Feature Request] Better error handling for VPN configuration errors
**Problem description**
Error handling of the vpn setup is very poor. If you make a mistake in either the server or node configuration file the final error will be an Authentication error, which is not the real problem.

**Desired solution**
Better error handling for the VPN certifical retrieval

### PR DESCRIPTION
**Pull request by bartvanb**
Fix #278 

In working on this, I also noticed a design error if someone defines a subnet in the node configuration file that doesn't match the subnet in which VPN IPs are provided. Then, there was no error of any kind, however the VPN connection would not have worked because the IP tables rules were generated for the wrong subnet. This has now been resolved with an extra check: when the VPN is connected, check if the VPN IP falls within the node subnet range; otherwise give an error and turn VPN off explicitly

### CODE REVIEW

File: vantage6-server/vantage6/server/resource/vpn.py [hunk 1]
```diff
@@ -381,18 +485,30 @@ def _remove_keypair(self, ovpn_config: str) -> str:
...
+        Returns
+        -------
+        Dict
+            Response content of the EduVPN /profile_list route
+        """
         response = self.session.get(f'{self.API_URL}/profile_list')
         return json.loads(response.content.decode('utf-8'))
 
-    def get_config(self, profile_id):
+    def get_config(self, profile_id) -> str:
```
Comment (frankcorneliusmartin) [inline vantage6-server/vantage6/server/resource/vpn.py:500]: missing type

File: vantage6-server/vantage6/server/resource/vpn.py [hunk 2]
```diff
@@ -311,14 +386,33 @@ def _authorize(self):
             params=params,
             allow_redirects=False
         )
+        if not (200 <= response.status_code < 400):
+            raise VPNPortalAuthException(
+                "Authenticating to EduVPN failed. Please check the "
+                "'client_id' and 'redirect_url' settings in your server "
+                "configuration file."
+            )
+        elif not response.headers.get('Location'):
```
Comment (frankcorneliusmartin) [inline vantage6-server/vantage6/server/resource/vpn.py:395]: `elif 'location' response.headers`

**Review by frankcorneliusmartin — APPROVED** [review]
lgtm. minor changes

File: vantage6-server/vantage6/server/resource/vpn.py (hunk 2 above)
Comment (bartvanb) [reply vantage6-server/vantage6/server/resource/vpn.py]: done

File: vantage6-server/vantage6/server/resource/vpn.py (hunk 1 above)
Comment (bartvanb) [reply vantage6-server/vantage6/server/resource/vpn.py]: done

**Pull request merged**